.github/workflows/nim_test.yml

```diff
@@ -0,0 +1,33 @@
+---
+name: nim_test
+
+# yamllint disable-line rule:truthy
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  nim_test:
+    runs-on: ${{matrix.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        nim-version:
+          - '1.6.12'
```

Comment:
Add devel 
```suggestion
          - '1.6.12'
          - 'devel'
```

Reply:
Do we want to require passing tests on devel?
Current guidelines mention it only as a recommendation

Reply:
What do you think about having an additional workflow similar to one [here](https://github.com/TheAlgorithms/Nim/pull/20#discussion_r1213929735) running both `stable` and `devel`? Then there would be a cyclic check if the code still builds with the newest versions, but would not possibly spoil some PRs.

Reply:
I don't mind having it, depends on what will trigger it.

Reply:
I want to check tests on devel, not require passing tests.
A devel is a future stable version, it helps knowing the changes we will have to bring.